pkg/chunk/azure/blob_storage_client.go

```diff
@@ -17,11 +17,25 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
+//blob service endpoint of AzureGlobal
 const blobURLFmt = "https://%s.blob.core.windows.net/%s/%s"
 const containerURLFmt = "https://%s.blob.core.windows.net/%s"
 
+// blob service endpoint of AzureChinaCloud
+const chinaBlobURLFmt = "https://%s.blob.core.chinacloudapi.cn/%s/%s"
+const chinaContainerURLFmt = "https://%s.blob.core.chinacloudapi.cn/%s"
+
+// blob service endpoint of AzureGermanCloud
+const germanyBlobURLFmt = "https://%s.blob.core.cloudapi.de/%s/%s"
+const germanyContainerURLFmt = "https://%s.blob.core.cloudapi.de/%s"
+
+// blob service endpoint of AzureUSGovernment
+const govBlobURLFmt = "https://%s.blob.core.usgovcloudapi.net/%s/%s"
+const govContainerURLFmt = "https://%s.blob.core.usgovcloudapi.net/%s"
+
 // BlobStorageConfig defines the configurable flags that can be defined when using azure blob storage.
 type BlobStorageConfig struct {
+	CloudEnvironment   string         `yaml:"cloud_environment"`
```

Comment:
What if we simply call it `environment`?

Reply:
@pracucci I can not follow cortexproject , do you have any ideas？

![Aug-08-2020 12-24-21](https://user-images.githubusercontent.com/42572980/89702344-2432d900-d973-11ea-8a1a-c96b30578ab7.gif)

Reply:
I think I've followed cortex, right？

Reply:
done